src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -113,39 +113,58 @@ private void WriteRequestCreation(CodeWriter writer, RestClientMethod operation)
 
                 writer.Line($"var {message:D} = _pipeline.CreateMessage();");
                 writer.Line($"var {request:D} = {message}.Request;");
-                var method = operation.Request.HttpMethod;
+                var method = clientMethod.Request.HttpMethod;
                 writer.Line($"{request}.Method = {typeof(RequestMethod)}.{method.ToRequestMethodName()};");
 
-                //TODO: Add logic to escape the strings when specified, using Uri.EscapeDataString(value);
-                //TODO: Need proper logic to convert the values to strings. Right now, everything is just using default ToString().
-                //TODO: Need logic to trim duplicate slashes (/) so when combined, you don't end  up with multiple // together
-
                 writer.Line($"var {uri:D} = new RawRequestUriBuilder();");
-                foreach (var segment in operation.Request.HostSegments)
+                Parameter? nextLinkParameter = parameters.FirstOrDefault(p => p.Name == "nextLink");
+                bool hasArtificialNextLink = nextLinkParameter?.IsArtificial == true;
+                PathSegment? nextLinkSegment = clientMethod.Request.PathSegments.FirstOrDefault(s => !s.Value.IsConstant && s.Value.Reference.Name == "nextLink");
```

Comment:
Let's mark the segment in the building code and not rely on the name here.

Reply:
Done. We only use `IsRaw` in PathSegment now for this determination.